app/javascript/components/map/VMapPASearch.vue

```diff
@@ -1,17 +1,88 @@
 <template>
-  <div>
-    Search
+  <div class="v-map-pa-search">
+    <input
+      ref="input"
+      class="v-map-pa-search__input"
+      type="text"
+      :placeholder="type.placeholder"
+      v-model="query"
+      @input="onInput"
+      @keyup.enter.prevent="onEnter"
+    />
+    <div class="v-map-pa-search__search_icon" @click="onIconClick" />
   </div>
 </template>
 
 <script>
+import axios from "axios";
+
 export default {
-  name: 'VMapPASearch',
+  name: "VMapPASearch",
+
+  model: {
+    prop: "query",
+    event: "search"
+  },
 
-  data () {
+  data() {
     return {
+      query: "",
+      autocompleteResults: [],
+    };
+  },
+
+  props: {
+    type: {
+      type: Object,
+      required: true,
+      validator: type => {
+        return type.hasOwnProperty('id') &&
+          type.hasOwnProperty('placeholder')
+      }
     }
   },
 
-}
+  computed: {
+    hasValidQuery() {
+      return this.query && this.query.length > 2;
+    }
+  },
+
+  methods: {
+    onInput(e) {
+      this.$emit("input", e.target.value);
+    },
+
+    focusInput() {
+      this.$refs.input.focus();
+    },
+
+    onIconClick() {
+      if (this.query) {
+        this.submitSearch();
+      } else {
+        this.focusInput();
+      }
+    },
```

Comment:
Might be worth checking if the autocomplete is meant to update automatically as you type. I suppose this depends how well it can cope with that many requests.

Reply:
I was planning on implementing this actually!

Reply:
Cool! Check with Ben or Ferdi first probably just to make sure the backend will support this fine. Also there is already a search comp in the site. Have you looked to see if any of that is reusable?